dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/icon/IconController.java

```diff
@@ -88,12 +102,20 @@ public class IconController {
   @GetMapping("/{iconKey}/icon.svg")
   public void getIconData(HttpServletResponse response, @PathVariable String iconKey)
       throws IOException, NotFoundException {
-    Resource icon = iconService.getDefaultIconResource(iconKey);
+    downloadDefaultIcon(iconKey, response);
+  }
 
-    response.setHeader("Cache-Control", CacheControl.maxAge(TTL, TimeUnit.DAYS).getHeaderValue());
-    response.setContentType(MediaType.SVG_UTF_8.toString());
+  @GetMapping(value = "/{key}/icon")
+  public void getCustomIconData(@PathVariable String key, HttpServletResponse response)
```

Comment:
The method names are confusing to me at least: there are default icons that come with DHIS2 and custom icons users can upload. This method serves both but is called `getCustomIconData`. The `/{iconKey}/icon.svg` endpoint is handled by `getIconData` but only returns default icons.

Is `/{iconKey}/icon.svg` supposed to only return default icons? Can custom icons also be stored as svgs? If `/{iconKey}/icon.svg` is only returning default icons the method should be renamed to something like `getDefaultIconData`.

`/{key}/icon` returns both types of icons while the method is called `getCustomIconData`. `getIconData` might be clearer 😅 .

Reply:
I agree with Ivo in that if the method returns both types of icons, it would make more sense to rename it to `getIconData`.

Since we have this new GET endpoint to download icons, seems like a smart decision to group both default and custom icons in here, but then, is it still useful to expose `/{iconKey}/icon.svg`?

Reply:
I agree, we can rename the existing to "getCoreIcon" or something like that, to specify the kind of icon it returns, while the new could be "getIcon", since it returns any icon.

We should keep the iconKey/icon.svg endpoint to be backwards compatible, but mark it as deprecated.

Reply:
@stian-sandvold did `iconKey/icon.svg` exist before? I thought this custom icon was all new 🤔

Reply:
fixed

Reply:
Custom icon is all new, but the iconKey/icon.svg is how we serve the bundled icons :)